From: Alejandro del Castillo <alejandro.delcastillo@ni.com>
Date: Mon, 15 May 2017 17:32:15 -0500
Subject: [PATCH] parse_userfields: avoid buffer overrun with user fields > 1023

User defined fields longer than 1023 bytes were causing a buffer
overrun since no max length was specified on sscanf. Increase the user
defined field maximum length to 4095 characters, and enforce a limit on
the sscanf, to avoid the overrun.

Signed-off-by: Alejandro del Castillo <alejandro.delcastillo@ni.com>
---
 libopkg/pkg_parse.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libopkg/pkg_parse.c b/libopkg/pkg_parse.c
index d2badd6..28a01a0 100644
--- a/libopkg/pkg_parse.c
+++ b/libopkg/pkg_parse.c
@@ -65,10 +65,10 @@ static void parse_conffiles(pkg_t * pkg, const char *cstr)
 
 static void parse_userfields(pkg_t *pkg, const char *cstr)
 {
-    char name[1024], value[1024];
+    char name[1024], value[4096];
     int r;
 
-    r = sscanf(cstr, "%1023s %[^\n]", name, value);
+    r = sscanf(cstr, "%1023s %4095[^\n]", name, value);
     if (r != 2) {
         opkg_msg(ERROR, "Failed to parse User Field line for %s\n", pkg->name);
         return;
-- 
2.7.4

